### PR TITLE
Sim 2381/light curve kwargs

### PR DESCRIPTION
**Pull request by danielsf**
This pull request makes `constraint` a kwarg of `LightCurveGenerator.light_curves_from_pointings()` and `boundLength` a kwarg of `LightCurveGenerator.get_pointings()`.

### CODE REVIEW

File: python/lsst/sims/catUtils/utils/LightCurveGenerator.py [hunk 1]
```diff
@@ -256,7 +263,7 @@ def get_pointings(self, ra, dec, bandpass=('u', 'g', 'r', 'i', 'z', 'y'), expMJD
                                                               fieldDec=dec,
                                                               telescopeFilter=bandpass,
                                                               expMJD=expMJD,
-                                                              boundLength=1.75)
+                                                              boundLength=boundLength)
```
Comment (sambolino) [inline python/lsst/sims/catUtils/utils/LightCurveGenerator.py]: there's one more boundLenght=1.75 at line 273/280

File: python/lsst/sims/catUtils/utils/LightCurveGenerator.py [hunk 2]
```diff
@@ -333,15 +333,28 @@ def _get_query_from_group(self, grp, chunk_size, lc_per_field=None):
...
 
+        if self._constraint is not None and constraint is not None:
+            master_constraint = self._constraint + ' AND ' + constraint
+        elif self._constraint is not None and constraint is None:
+            master_constraint = self._constraint
+        elif self._constraint is None and constraint is not None:
+            master_constraint = constraint
+        else:
+            master_constraint = None
+
```
Comment (sambolino) [inline python/lsst/sims/catUtils/utils/LightCurveGenerator.py]: this is ok but just for fun
```
#convert None to empty string
none_to_empty = lambda s: s or ""
#concatenate and remove potential loose ' AND ' from both ends
master_constraint = ' '.join([none_to_empty(self._constraint), 'AND', none_to_empty(constraint)]).strip(' AND ')
```

File: python/lsst/sims/catUtils/utils/LightCurveGenerator.py (hunk 1 above)
Comment (danielsf) [inline python/lsst/sims/catUtils/utils/LightCurveGenerator.py]: Thanks.  Fixed it.

File: python/lsst/sims/catUtils/utils/LightCurveGenerator.py (hunk 2 above)
Comment (danielsf) [inline python/lsst/sims/catUtils/utils/LightCurveGenerator.py]: I'm going to leave the `if/elif` block, as I think that will be easier to parse for anyone who has to fix this code in the future.

**Comment by danielsf** [comment]
Thanks for the review.  I will test this on Jenkins and then merge it.

**Pull request merged**